cmd/internal/ec2/add.go

```diff
@@ -0,0 +1,132 @@
+package ec2
+
+import (
+	"context"
+	"os/exec"
+
+	// "github.com/AlecAivazis/survey/v2"
+	"github.com/AlecAivazis/survey/v2/terminal"
+	// "github.com/akitasoftware/akita-cli/cfg"
+	"github.com/akitasoftware/akita-cli/cmd/internal/cmderr"
+	"github.com/akitasoftware/akita-cli/printer"
+	"github.com/akitasoftware/akita-cli/telemetry"
+	"github.com/akitasoftware/go-utils/optionals"
+	"github.com/pkg/errors"
+)
+
+// Helper function for reporting telemetry
+func reportStep(stepName string) {
+	telemetry.WorkflowStep("Add to EC2", stepName)
+}
+
+// A function which executes the next part of the workflow,
+// and picks a next state (Some) or exits (None), or signals an eror.
+type AddWorkflowState func(*AddWorkflow) (next optionals.Optional[AddWorkflowState], err error)
+
+// Helper functions for choosing the next state.
+func awf_done() (optionals.Optional[AddWorkflowState], error) {
+	// I do not understand why Go cannot infer this type.
+	return optionals.None[AddWorkflowState](), nil
+}
+
+func awf_error(err error) (optionals.Optional[AddWorkflowState], error) {
+	return optionals.None[AddWorkflowState](), err
+}
+
+func awf_next(f AddWorkflowState) (optionals.Optional[AddWorkflowState], error) {
+	return optionals.Some[AddWorkflowState](f), nil
+}
```

Comment:
All this was only necessarily in the interactive version because we allowed people to backup or retry a selection.  In the simplified workflow, there is no retry on error or no prompting, so why keep all this machinery?